packages/compass/src/main/application.ts

```diff
@@ -54,6 +62,34 @@ class CompassApplication {
     this.setupLifecycleListeners();
     this.setupApplicationMenu();
     this.setupWindowManager();
+
+    const {
+      protectConnectionStrings,
+      readOnly,
+      file,
+      positionalArguments,
+      autoUpdates,
+      // TODO: COMPASS-6063
+      // maxTimeMS,
+    } = preferences.getPreferences();
+
+    debug('application launched');
+    track('Application Launched', {
+      context: mode,
+      launch_connection: launchConnection(file, positionalArguments),
+      protected: protectConnectionStrings,
+      readOnly,
+      autoUpdates,
+      // TODO: COMPASS-6063
+      // maxTimeMS,
+      has_global_config: !!Object.keys(globalPreferences.global).length,
+      has_cli_config: !!Object.keys(globalPreferences.cli).length,
+    });
```

Comment:
Do you want to split this out into a separate function, like the ones above?

Reply:
(again, not sure why this conversation was resolved – not a big deal and definitely not something blocking for this PR, but since you gave this a :+1: reaction, I assumed that you actually did want to make this change, so something’s messed up with your git client again maybe? :confused:)

Reply:
This time I am pretty sure I changed it! :)

https://github.com/mongodb-js/compass/blob/COMPASS-6077-track-application-launched/packages/compass/src/main/application.ts#L93-L94

Reply:
<img width="728" alt="Screenshot 2022-11-17 at 11 55 25" src="https://user-images.githubusercontent.com/16307679/202428138-8e63f88d-de09-49d5-b323-0ceb4d455c0a.png">

Reply:
Ah sorry, I was referring to this entire block, both the `getPreferences()` and the `track()` call, which could be treated like the functions immediately above this block (the `this.setup...()` ones), but yeah, again, not important or blocking, so if you’re good with this I am too :slightly_smiling_face:

Reply:
Aaaaah got it! Yeah, makes sense, something like `this.trackApplicationLaunched()`.